rfcs/text/0000-bundle-definition.md

```diff
@@ -0,0 +1,248 @@
+
+1. As an Azul or Query Service developer I want to be able to distinguish between different types of bundles in order to
+   know which bundle to index, link, etc. or to provide bundle types to users as a search criterion.
+
+1. As a computational biologist, I would like to use the same references that the HCA DCP uses in their analysis.
+
+## Detailed Design
+
+### What is a bundle?
+
+A bundle is a unit of data organization, roughly equivalent to a folder or directory. In the HCA DCP, a bundle is
+defined as either a transactional grouping of files that belong together for scientific reasons, or a useful association
+between a set of files for utility purposes.
+
+* Data bundles contain the results of a single data transaction. They can be viewed as the digital equivalent of an
+  "instrument run" - a quantity of data gathered from a single process instance. Data bundles are annotated with
+  metadata using a procedure designed by the metadata team.
+
+* Utility bundles exist to organize (group) data bundles and represent data of non-DCP origin (such as reference data)
+  that is used in the course of analysis.
+
+### How are bundle types defined?
+
+Operationally, bundle types can be defined like file types or media types: they are used as a data type identifier for
+applications to pattern match on whether data should be considered as suitable for input to the application.
+
+### Bundle types and their definitions
+
+Bundle types, definitions, consumers, and specifications are outlined below. Use cases from above are indicated where
+supported by a specific bundle type.
+
+#### Table header definitions
+
+- **Type**: The name of the bundle type
+
+- **Definition**: What is and is not in this bundle type
+
+- **Target consumer**: Who will be consuming this bundle type
+
+- **Use cases**: Which Users Stories are supported by this bundle type
+
+| Type | Definition | Target consumer | Use cases |
+|:-|:-|:-|:-|
+| Primary bundle | A bundle that contains primary (raw) data files and all related metadata files. | computational biologists; Data Processing Pipelines; Data Browser; Query Service | 1 |
+| Secondary bundle | A bundle that contains secondary (alignment, expression) data files produced by the Data Processing Pipelines, input primary data files, and all related metadata files. | computational biologists; Data Processing Pipelines; Data Browser; Query Service; Matrix Service | 2, 3, 4(?) |
+| Tertiary bundle* | A bundle that contains tertiary (expression) data files produced by the Matrix Service, input primary and secondary data files, and all related metadata files. | computational biologists; biologists; Data Browser; Query Service | 3, 4 |
+| Project bundle* | A bundle that contains project-level metadata files. This bundle type does not contain data files or metadata files related to biomaterials, protocols, processes, or files. | computational biologists; biologists; | ? |
```

Comment:
What is the rational for the "Project Bundle" What use case etc was thais meant to accomplish or what task made easier. Why is this being proposed? Thanks!

Reply:
That is described in the *New Project bundle* section below.